oracle/gasPriceOracle.go

```diff
@@ -5,35 +5,111 @@ package oracle
 
 import (
 	"github.com/ChainSafe/sygma-fee-oracle/types"
+	"github.com/ChainSafe/sygma-fee-oracle/util"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	"math/big"
 )
 
 type GasPriceOracle interface {
 	Oracle
-	InquiryGasPrice(chainDomain string) (*types.GasPrices, error)
+	InquiryGasPrice() (*types.GasPrices, error)
 }
 
+type GasPriceConverter func(gasPrices *types.GasPrices) (*types.GasPrices, error)
+
 type GasPriceOracleOperator struct {
 	log *logrus.Entry
 
 	oracle GasPriceOracle
+
+	gasPriceConverterFns map[string]GasPriceConverter
 }
 
 func NewGasPriceOracleOperator(log *logrus.Entry, oracle GasPriceOracle) *GasPriceOracleOperator {
 	return &GasPriceOracleOperator{
 		log:    log.WithField("oracle", "gas price"),
 		oracle: oracle,
+		gasPriceConverterFns: map[string]GasPriceConverter{
+			"etherscan":   etherscanGasPriceConverter,
+			"polygonscan": polygonscanGasPriceConverter,
+			"moonscan":    moonscanGasPriceConverter,
+		},
 	}
 }
 
-func (g *GasPriceOracleOperator) Run(chainDomain string) (*types.GasPrices, error) {
-	return g.oracle.InquiryGasPrice(chainDomain)
+func (g *GasPriceOracleOperator) Run() (*types.GasPrices, error) {
+	gasPriceData, err := g.oracle.InquiryGasPrice()
+	if err != nil {
+		return nil, err
+	}
+	return g.GasPriceUnitConverter(gasPriceData)
 }
 
-func (g *GasPriceOracleOperator) GetOracleName() string {
-	return g.oracle.Name()
+func (g *GasPriceOracleOperator) GetOracleSource() string {
+	return g.oracle.Source()
 }
 
 func (g *GasPriceOracleOperator) IsOracleEnabled() bool {
 	return g.oracle.IsEnabled()
 }
+
+func (g *GasPriceOracleOperator) GetOracle() GasPriceOracle {
+	return g.oracle
+}
+
+func (g *GasPriceOracleOperator) GasPriceUnitConverter(gasPrices *types.GasPrices) (*types.GasPrices, error) {
+	return g.gasPriceConverterFns[g.oracle.Source()](gasPrices)
+}
+
+func etherscanGasPriceConverter(gasPrices *types.GasPrices) (*types.GasPrices, error) {
```

Comment:
is it necessary to have denomination? because the gas price denomination will barely change for each oracle service, when impl a new oracle, we def know what it is, so adding a converter func in a mapping seems to be a good option to me. this way we have flexibility to handle it in this `GasPriceConverter` logic layer, in case there are some weird gas price api response